"Verify SSL Cert" added as option to fix broken SSL installs

theguardian opened the pull request:
Appliances like QNAP, with their latest firmware, have completely broken SSL / python integration.

urllib2 with python 2.7.10 (i think?) requires valid ssl certs on a machine to make routine calls. This new configurable setting allows the user to bypass the new required behavior so they don't have a broken experience.

For me, this fixed twitter notifications - but this could also fix any https:// hits for others in a similar situation.

basilfx commented:
For the reason(s) explained in the [troubleshooting wiki](https://github.com/rembo10/headphones/wiki/Troubleshooting#ssl-enabled-serversindexerssearch-providers-dont-work), I would not prefer to have an option in the user interface to switch off SSL certificate checks.

theguardian commented:
agree on your comments. patch is needed (in addition to already existing code) only when python version > 2.7.9. Removed web interface to toggle and moved the patch code over to requests.py

SickRage handles it like a global monkeypatch: https://github.com/SiCKRAGETV/SickRage/blob/4d37c1352e1a27023ffd520a4da8ab0f024333e6/SickBeard.py#L51

There's a better way to do it: https://www.python.org/dev/peps/pep-0476/ but I don't think all of HP functionality handles urllib2 the same way (e.g. pythontwitter package). I think this is a sufficient stop-gap for now.
